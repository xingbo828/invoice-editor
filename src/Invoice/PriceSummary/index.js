import { compose, withProps } from 'recompose'
import { connect } from 'react-redux'

import { getProductsTotal } from '../../cartReducer'


import PriceSummary from './PriceSummary'

const tax = 0.05

const formatter = new Intl.NumberFormat('en-US', {
  style: 'currency',
  currency: 'USD',
  minimumFractionDigits: 2
})

const mapStateToProps = (state) => ({
  subtotal: getProductsTotal(state)
});

const enhance = compose(
  connect(mapStateToProps),
  withProps(({ subtotal }) => ({
    tax: tax * subtotal,
    total: subtotal * (1 + tax),
    formatter
  }))
)


export default enhance(PriceSummary)