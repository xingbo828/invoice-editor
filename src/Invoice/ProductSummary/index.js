import { compose, withHandlers } from 'recompose'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import { updateProduct } from '../../cartAction'
import { getProducts } from '../../cartReducer'


import ProductSummary from './ProductSummary'

const mapDispatchToProps = dispatch => bindActionCreators({
  updateProduct
}, dispatch)

const mapStateToProps = (state) => ({
  products: getProducts(state)
});

const enhance = compose(
  connect(mapStateToProps, mapDispatchToProps),
  withHandlers({
    handleRemove: props => (product) => e => {
      props.updateProduct({
        product,
        quantity: 0
      })
    }
  })
)


export default enhance(ProductSummary)