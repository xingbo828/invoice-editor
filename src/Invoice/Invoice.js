import React from 'react';

import Container from '../Shared/Container'
import Heading from '../Shared/Heading'

import ProductSummary from './ProductSummary'
import PriceSummary from './PriceSummary'

const Invoice = () => {
  return (
    <Container>
      <Heading wrapperTag="h1" centered size="md">Your invoice</Heading>
      <ProductSummary />
      <PriceSummary />
    </Container>
  );
};

export default Invoice;