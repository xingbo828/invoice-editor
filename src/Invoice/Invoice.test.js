import React from 'react';
import { shallow } from 'enzyme';
import Invoice from './Invoice';

it('should match snapshot', () => {
  const wrapper = shallow(<Invoice />);
  expect(wrapper).toMatchSnapshot()
});