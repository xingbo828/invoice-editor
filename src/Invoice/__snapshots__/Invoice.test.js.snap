// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Invoice />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef
          centered={true}
          size="md"
          wrapperTag="h1"
        >
          Your invoice
        </ForwardRef>,
        <Connect(withHandlers(ProductSummary)) />,
        <Connect(withProps(PriceSummary)) />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "centered": true,
          "children": "Your invoice",
          "size": "md",
          "wrapperTag": "h1",
        },
        "ref": null,
        "rendered": "Your invoice",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "WithTheme(MarketingHeading)",
          "render": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  padding: ",
          [Function],
          ";
  margin: ",
          [Function],
          ";
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            centered={true}
            size="md"
            wrapperTag="h1"
          >
            Your invoice
          </ForwardRef>,
          <Connect(withHandlers(ProductSummary)) />,
          <Connect(withProps(PriceSummary)) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "centered": true,
            "children": "Your invoice",
            "size": "md",
            "wrapperTag": "h1",
          },
          "ref": null,
          "rendered": "Your invoice",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "WithTheme(MarketingHeading)",
            "render": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "rules": Array [
            "
  padding: ",
            [Function],
            ";
  margin: ",
            [Function],
            ";
  ",
            [Function],
            "
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
